### PR TITLE
tests: accelerate and fix integration tests

### PR DESCRIPTION
**Pull request by UniqMartin**
Speed-up is achieved by avoiding to generate the coverage report in `index.html` on *every run* of a command in the integration tests (this makes those about 1/3 faster).

Fix means generate unique names for the commands in the cached result set created by SimpleCov, so that multiple commands with identical arguments don't overwrite each other's coverage results.

cc @bfontaine

### CODE REVIEW

**Comment by xu-cheng** [comment]
:+1:

File: Library/Homebrew/test/.simplecov [hunk 1]
```diff
@@ -14,14 +14,23 @@ SimpleCov.start do
...
+    end
+  end
+end
+
+if ENV["HOMEBREW_INTEGRATION_TEST"]
+  SimpleCov.command_name ENV["HOMEBREW_INTEGRATION_TEST"]
   SimpleCov.at_exit do
     exit_code = $!.nil? ? 0 : $!.status
     $stdout.reopen("/dev/null")
     SimpleCov.result.format!
```
Comment (bfontaine) [inline Library/Homebrew/test/.simplecov]: Isn’t it simpler to remove this `.format!` call? Calling `.result` should be enough to force SimpleCov to merge results.

Comment (UniqMartin) [inline Library/Homebrew/test/.simplecov]: You're right! Calling `.result` is already sufficient and there's no need for a dummy formatter. Short and simple solution. :smiley_cat:

**Comment by bfontaine** [comment]
:+1:

**Pull request closed without merging**